iati/home/models.py

```diff
@@ -117,12 +118,13 @@ def get_context(self, request, *args, **kwargs):
 
     def clean(self):
         """Override clean to remove trailing dashes from slugs with whitespaces."""
-        for lang in ['en', 'fr', 'es', 'pt']:
+        for lang in tuple(x[0] for x in settings.LANGUAGES):
             slug_field = 'slug_{}'.format(lang)
             slug = getattr(self, slug_field)
             if slug:
-                slug_clean = re.sub(r'[^\w\s-]', '', slug).strip("-").lower()
-                setattr(self, slug_field, slug_clean)
+                slug_stripped = re.sub(r'[^\w\s-]', '', slug).strip("-")
+                slug_valid = slugify(slug_stripped)
```

Comment:
@Ocre42 

This will verify that the slug is a valid slug - previously `slug_clean` was converted to lowercase, but best to be safe